include/libaries_askar.h

```diff
@@ -362,7 +368,13 @@ ErrorCode askar_key_from_seed(FfiStr alg,
                               FfiStr method,
                               LocalKeyHandle *out);
 
-ErrorCode askar_key_generate(FfiStr alg, int8_t ephemeral, LocalKeyHandle *out);
+/**
+ * id is an optional argument only required for hardware-bound keys
+ */
+ErrorCode askar_key_generate(FfiStr alg,
+                             int8_t hardware_backed,
```

Comment:
Instead of marking something as hardware_backed, should this be a more generic parameters so it's a bit more ready for future changes? E.g. "key_backend" which can take a value of `secure_env`, or something like `strongbox`? I foresee us using the same capabilities for cloud HSM, with possibly multiple backends configured, and it might be nice to make it a backend rather than strictly hardware_backed?

Just a thought, this approach seems fine as well